overlord/snapshotstate/backend/backend.go

```diff
@@ -100,13 +102,26 @@ func Iter(ctx context.Context, f func(*Reader) error) error {
 				break
 			}
 
+			// filter out non-snapshot directory entries
+			ok, setID := isSnapshotFilename(name)
+			if !ok {
+				continue
+			}
 			filename := filepath.Join(dirs.SnapshotsDir, name)
 			reader, openError := backendOpen(filename)
 			// reader can be non-nil even when openError is not nil (in
 			// which case reader.Broken will have a reason). f can
 			// check and either ignore or return an error when
 			// finding a broken snapshot.
 			if reader != nil {
+				if reader.SetID != setID && openError == nil {
+					// ignore snapshots where set id of the filename disagree
+					// with internal set id. This may be the case in the future
+					// with new enhanced snapshot format.
+					// XXX: warning instead?
+					logger.Debugf("Snapshot %q ignored, internal set-id %d disagrees with filename", name, reader.SetID)
```

Comment:
the code used the spelling "set id", no hyphen, let's be consistent

Reply:
this should be a Noticef I think

Reply:
Heh, I think we asked him to change it. Oh well, it can be changed back, no big deal. But I really think we should not make this a warning, I would remove the "XXX"